.github/workflows/ci.yaml

```diff
@@ -29,7 +29,7 @@ jobs:
         with:
           path: "oci_env"
 
-      - uses: actions/setup-python@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: "3.8"
```

Comment:
Would it be possible to use a newer version of Python?

Reply:
Yes, on the CI we can use newer version of python more freely, because the python used by pulp itself is defined in the container image (pulp-oci-images repo).

I think it's safe to stick with what is in pulpcore, for example: https://github.com/pulp/pulpcore/blob/7185bdf32cf0e4eeccee1b5d2f93b65174f53331/.github/workflows/build.yml#L33

And just an eagle view on our CI: core and plugins (rpm, deb, etc) share the same base CI code, because we do have a CI templating system in-place to manage the plugins CI (core, rpm, deb, etc). I wouldnt bother with that just now, but it's the [plugin_template repo](https://github.com/pulp/plugin_template/), case you didn't heard of it already. But not all repositories have their CI manage, like this one, openapi-generator, cli, docs.